repos/keg-components/src/components/textToggle/textToggle.js

```diff
@@ -0,0 +1,179 @@
+import React, { useState, useMemo, useCallback } from 'react'
+import { Text, Touchable, Drawer } from '../'
+import { View } from 'KegView'
+import { useClassList } from 'KegClassList'
+import { isValidComponent } from '../../utils'
+import PropTypes from 'prop-types'
+import { useStylesCallback } from '@keg-hub/re-theme'
+import LinearGradient from 'react-native-linear-gradient'
+import { isFunc } from '@keg-hub/jsutils'
+
+/**
+ * build the styles object based on togglePosition
+ * @param {object} theme
+ * @param {object} styleHelper - of the form { togglePosition, styles }
+ */
+const buildStyles = (theme, styleHelper) => {
+  const textToggleStyles = theme.get(`textToggle`, styleHelper?.styles)
+
+  // default right position
+  let align = 'flex-end'
+  switch (styleHelper?.togglePosition) {
+  case 'left':
+    align = 'flex-start'
+    break
+  case 'center':
+    align = 'center'
+    break
+  }
+  return theme.get(textToggleStyles, {
+    main: {
+      alignItems: align,
+    },
+  })
+}
+
+/**
+ * TextToggle
+ * @param {object} props
+ * @param {string} props.text - text to display as the content
+ * @param {object=} props.styles
+ * @param {boolean=} props.isExpandedInit - whether the content is expanded or not initially
+ * @param {string=} props.className
+ * @param {Function=} props.onToggleChange - optional. callback whenever the toggle component is pressed
+ * @param {'left'|'center'|'right'=} props.togglePosition - optional. where to position the toggle component. default 'right'
+ * @param {Number=} props.collapsedHeight - optional. height of the textview when collapsed
+ * @param {Component=} props.CustomToggle - optional toggle component to use instead of the default
+ * @param {string} props.fadeColor - optional color for the fade on collapsed text
+ */
+export const TextToggle = props => {
+  const {
+    text,
+    styles,
+    isExpandedInit = false,
```

Comment:
This seems overly verbose, just to say `expanded`

Reply:
i want to signify that setting that will not change the toggled state after the first render